File: larry_the_zookeeper.py
import time, sys
class Larry():
    def __init__(self):
        pass

    def delay(self):
        print()

def format_quote(quote): #trying to figure out how i would use this
    return quote[0] + ': ' + quote[1]
        
def store_file(filename):
    file = open(filename)#directory
    file_content = file.readlines()
    dialog = []
    
    for line in file_content:
        ind = line.index(':')
        dialog.append( (line[0:ind].strip(), line[ind+1:].strip()) )
    return dialog

def print_delayed_text(text, delay):
    for letter in text:
        print(letter,end='')
        sys.stdout.flush()
        time.sleep(delay)

def print_quote(quote):
    print_delayed_text(quote[0] + ': ', .05)
    print_delayed_text(quote[1] + '\n', .2)

def main():
    if len(sys.argv) != 2:
        print("You must provide a filename my friend.")
        return
    
    dialog = store_file(sys.argv[1])
    
    for quote in dialog:
        print_quote(quote)
    
if __name__ == '__main__':
    main()

File: larry_the_zookeeper.go
package main

import (
	"time"
	"fmt"
	"os"
	"io"
	"bufio"
)

func load_file(filename string) (s string) {
	file, err := os.Open(filename)
	if err != nil {
		panic(err)
	}

	defer func() {
		if err := file.Close(); err != nil {
			panic(err)
		}
	}()

	r := bufio.NewReader(file)

	buf := make([]byte, 1024)
	for {
		n, err := r.Read(buf)
		if err != nil && err != io.EOF {
			panic(err)
		}

		if n == 0 { break }
		s += string(buf[:n])
	}

	return s
}

func print_delayed_text(text string, delay time.Duration) {
	tlen := len(text)
	for i := 0; i < tlen; i++ {
		fmt.Print(string(text[i]))
		time.Sleep(delay * time.Millisecond)
	}
}

func main() {
	if l := len(os.Args); l == 1 {
		fmt.Println("You must provide a filename my friend.")
		return
	} else if l > 2 {
		fmt.Println("Sorry, I only work with one file.")
		return
	}

	dialog := load_file(os.Args[1])
	print_delayed_text(dialog, 1)
}
